features/environment.py

```diff
@@ -640,7 +640,8 @@ def create_instance_with_uat_installed(
         )
         return
 
-    user_data = _get_user_data_for_instance(context.config, series)
+    if not user_data:
+        user_data = _get_user_data_for_instance(context.config, series)
```

Comment:
could this interfere with how we're installing `pro` from -proposed or the daily ppa?

Reply:
I don't think so, because this `user_data` in only being passed by the step that creates the machine. Only on the new integration test, this variable will contain something that is not `None`. But let me know if I am missing something here

Reply:
Right, but will we be able to run that new integration test against -proposed if the user_data is overwritten?